part-2.py

```diff
@@ -4,13 +4,52 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query):
+    if len(array)==0:
+        return False
+    elif array[-1]==query:
+        return True
+    else:
+        return search(array[:-1],query)
 
 
 # is_palindrome
-
+def is_palindrome(text):
+    if len(text) == 1 or len(text) == 0:
+        return True
+    elif text[0]==text[-1]:
+        return is_palindrome(text[1:-1])
+    else:
+        return False 
 
 
 # digit_match
-
+def digit_match(apples,oranges):
+    apples_list=[a for a in str(apples)]# Gives a list of ints
+    oranges_list=[o for o in str(oranges)]# Gives a list of ints
+    
+    #This section trims the list to be the same size
+    new_apples_list, new_oranges_list=trim_list(apples_list, oranges_list)
```

Comment:
Could this be handled by a base case instead?  For example, if the length of one list is 0 and the length of the other list is longer than zero, there are no more possible matches.